### PR TITLE
Documentation for deploying to internal storage.

### PR DESCRIPTION
**Pull request by igor-stoppa**
This section shows how to deploy an Ostro OS image
to the internal storage that might be available on certain
platforms.

Signed-off-by: Igor Stoppa <igor.stoppa@intel.com>

[skip-ci]

### CODE REVIEW

File: doc/howtos/booting-and-installation.rst [hunk 1]
```diff
@@ -127,6 +127,62 @@ includes a checksum verification, you can also use the traditional :command:`dd`
...
+=====================================
+   Ostro OS uses GID for identifying the rootfs and the GID is kept consistent across all the
+   images produced. This means that trying to boot a system with 2 Ostro OS images available at the
+   same time will likely produce unwanted/undetermined results.
+   Before trying again to boot from a removable media, the internal media should be wiped.
+   To achieve this:
+#. Boot from the internal media.
+#. Identify the root block device containing the rootfs partition.
+#. Wipe a sufficient part of that block device to make it unbootable.
+   In practice it means wiping the primary GPT and at least the beginngin of both partitions that
```
Comment (gvancuts) [inline doc/howtos/booting-and-installation.rst]: s/beginngin/beginning

**Comment by kad** [comment]
lsblk can be used to find available storage devices on the system, if we want to hint user somehow.

**Comment by pohly** [comment]
@gvancuts do you think that this can be merged now? You gave a "thumbs up" to kad's enhancement proposal, which might be understood as "please add that before merging".

**Comment by gvancuts** [comment]
Yes, that's what I meant with my thumbs up to @kad 's comment. I would like to recommend using lsblk.

**Comment by pohly** [comment]
@igor-stoppa do you agree to document lsblk as suggested by @kad and @gvancuts?

**Comment by igor-stoppa** [comment]
Not particularly, I would prefer to have the user to check some more authoritative source than lsblk. But I'm tired of discussing it, so I added it, as last resort.

**Comment by igor-stoppa** [comment]
@pohly btw, rebase -i origin/master worked, thanks!

File: doc/howtos/booting-and-installation.rst [hunk 2]
```diff
@@ -127,6 +127,73 @@ includes a checksum verification, you can also use the traditional :command:`dd`
...
+   be ``/dev/sdb``.
+#. Identify the block device representing the internal storage.
+   This part is really specific to each platfrom: typically the internal storage is
+   associated to ``/dev/sda`` for hard disks and SSD units and to ``/dev/mmcblk0`` for eMMC
+   devices, but it should be confirmed against the board manual or the BIOS, if it provides
+   such type of information.  Lacking that, one can run::
+
+      # lsblk
+
+   and try to interprete its output, to figure out which device represents the internal storage.
```
Comment (pohly) [inline doc/howtos/booting-and-installation.rst]: s/interprete/interpret/

**Comment by pohly** [comment]
@igor-stoppa thanks for accommodating the review feedback. Just one minor typo introduced by that change, though.

Good to hear that ```git rebase -i origin/master``` worked as intended :-) In simple cases like this, ```git commit --amend -a``` to include modifications directly in the existing commit is easier, though.

**Comment by igor-stoppa** [comment]
ok, done, it's probably time that I look for a spellchecker for vim :-/

**Comment by gvancuts** [comment]
Thanks for adding this @igor-stoppa . I understand your desire to add an 'authoritative' source of info. I just don't recall ever seeing a board manual or bios telling me what device name will be assigned to any component in Linux. lsblk is for all intent and purposes a more authoritative (and more readily available) source of information.

**Comment by pohly** [comment]
Looks the discussion has been settled, so I'll merge.

File: doc/howtos/booting-and-installation.rst [hunk 3]
```diff
@@ -127,6 +127,73 @@ includes a checksum verification, you can also use the traditional :command:`dd`
...
+
+   Shows that the root partition is on ``/dev/sdb3``, therefore the source block device will
+   be ``/dev/sdb``.
+#. Identify the block device representing the internal storage.
+   This part is really specific to each platfrom: typically the internal storage is
+   associated to ``/dev/sda`` for hard disks and SSD units and to ``/dev/mmcblk0`` for eMMC
+   devices, but it should be confirmed against the board manual or the BIOS, if it provides
+   such type of information.  Lacking that, one can run::
+
+      # lsblk
```
Comment (barbieri) [inline doc/howtos/booting-and-installation.rst]: sorry to be late for this review and whole discussion, but the `lsblk` allows ways to only show the actual device, not each partition, as well as limit the block type (to be only disks, etc). Last but not least one can show some properties (ie: hotpluggable, mount path...) and filter with grep. I've shared the command lines many times, but now I can't try one due lack of dev machine :-)

**Pull request merged**